src/cmds.c

```diff
@@ -2595,38 +2595,33 @@ static void cmd_pls_ignore(struct userrec *u, int idx, char *par)
 
   who = newsplit(&par);
   if (par[0] == '%') {
-    char *p, *p_expire;
-    unsigned long int expire_foo;
-
     p = newsplit(&par);
     p_expire = p + 1;
     while (*(++p) != 0) {
       switch (tolower((unsigned) *p)) {
       case 'd':
         *p = 0;
         expire_foo = strtol(p_expire, NULL, 10);
```

Comment:
I realize it has always been this way, but we should probably use a signed long there, %-1d shouldn't be valid (and not wrap around)